ci/inspect_changes.sh

```diff
@@ -0,0 +1,160 @@
+#!/bin/bash
+
+# Github action script to identify which subprojects are dirty in a PR
+
+set -u
+
+head_sha=$GITHUB_SHA
+base_sha=$BASE_SHA
+
+# Github gives the SHA as the current HEAD of the target ref, not the common ancestor.
+# Find the common ancestor and use that for the base.
+git fetch origin --unshallow -q
+git fetch origin $base_sha -q
+base_sha=$(git merge-base $head_sha $base_sha)
+
+# Define a list of subproject directories:
+subprojects=(
+  libcudacxx
+  cub
+  thrust
+)
+
+# ...and their dependencies:
+libcudacxx_dependencies=(
+  cccl
+)
+
+cub_dependencies=(
+  cccl
+  libcudacxx
+  thrust
+)
+
+thrust_dependencies=(
+  cccl
+  libcudacxx
+  cub
+)
```

Comment:
bash has associative arrays which can simplify this logic a bit. (And avoids the `eval` later in `add_dependencies()`.)

```suggestion
# ...and their dependencies:
declare -A dependencies=(
  [libcudacxx]="cccl"
  [cub]="cccl libcudacxx thrust"
  [thrust]="cccl libcudacxx cub"
)
```

Reply:
Great idea!